### PR TITLE
Add description for getTags()

### PR DESCRIPTION
**Pull request by mugdhalakhani**
@beverloo @rayankans PTAL.

### CODE REVIEW

**Comment by beverloo** [comment]
Please rebase this on top of the parent CL, as it now includes both.
[ML] Done!

File: spec/PeriodicBackgroundSync-index.bs [hunk 1]
```diff
@@ -151,3 +151,9 @@ The <code><dfn method for=PeriodicSyncManager title="register(tag, options)">reg
             1. Set |currentRegistration|'s associated [=options=] to |options|.
             1. Calculate an appropriate <var>timeToFire</var> for |currentRegistration| based on the [=minInterval=] value of |options|, and any other considerations the user agent has implemented per recommendations in [[#privacy]] and Resource Usage. The time interval between now and |timeToFire| MUST be greater than or equal to the [=minInterval=] specified in |options|. Set the |currentRegistration|'s [=time to fire=] to |timeToFire|.
     1. [=Resolve=] |promise|.
+
+The <code><dfn method for=PeriodicSyncManager title="getTags()">getTags()</dfn></code> method when invoked, MUST return [=a new promise=] <var>promise</var> and run the following steps [=in parallel=]:
+1. Let <var>serviceWorkerRegistration</var> be the {{PeriodicSyncManager}}'s associated [=service worker registration=].
+1. Let <var>currentTags</var> be a new [=sequence=].
```
Comment (beverloo) [inline spec/PeriodicBackgroundSync-index.bs]: Wouldn't we want to use `list` here?

Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs]: Did you mean any of these lists:
spec:url; type:dfn; text:list
spec:infra; type:dfn; text:list
Or something else? I've left it unlinked for now.

Comment (beverloo) [reply spec/PeriodicBackgroundSync-index.bs]: Infra's list.

Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs]: done.

**Comment by mugdhalakhani** [comment]
Gentle ping!

**Review by beverloo — APPROVED** [review]

File: spec/PeriodicBackgroundSync-index.bs [hunk 2]
```diff
@@ -24,6 +24,7 @@ spec:web-background-sync;
     type:dfn; text:online
     type:dfn; text:in the background
 spec:permissions-1; type:dict-member; text:name
+spec:infra; type:dfn; text:list
```
Comment (rayankans) [inline spec/PeriodicBackgroundSync-index.bs]: you don't need this if you just use list as: [=/list=]

File: spec/PeriodicBackgroundSync-index.bs [hunk 3]
```diff
@@ -204,3 +205,9 @@ The <code><dfn method for=PeriodicSyncManager title="register(tag, options)">reg
         1. Set |currentRegistration|'s associated [=state=] to [=pending=].
     1. Else, if |serviceWorkerRegistration|'s [=registration state=] is [=firing=], set |serviceWorkerRegistration|'s [=registration state=] to [=reregisteredWhileFiring=].
     1. [=Resolve=] |promise|.
+
+The <code><dfn method for=PeriodicSyncManager title="getTags()">getTags()</dfn></code> method when invoked, MUST return [=a new promise=] |promise| and run the following steps [=in parallel=]:
+1. Let |serviceWorkerRegistration| be the {{PeriodicSyncManager}}'s associated [=service worker registration=].
+1. Let |currentTags| be a new [=list=].
+1. For each |registration| in |serviceWorkerRegistration|'s [=list of periodicsync registrations=], add |registration|'s [=tag=] to |currentTags|.
```
Comment (rayankans) [inline spec/PeriodicBackgroundSync-index.bs]: |registration| of |serviceWorkerRegistration|'s ...

`in` does a different thing

Comment (rayankans) [inline spec/PeriodicBackgroundSync-index.bs]: add -> [=list/append=]

File: spec/PeriodicBackgroundSync-index.bs [hunk 4]
```diff
@@ -204,3 +205,9 @@ The <code><dfn method for=PeriodicSyncManager title="register(tag, options)">reg
         1. Set |currentRegistration|'s associated [=state=] to [=pending=].
     1. Else, if |serviceWorkerRegistration|'s [=registration state=] is [=firing=], set |serviceWorkerRegistration|'s [=registration state=] to [=reregisteredWhileFiring=].
     1. [=Resolve=] |promise|.
+
+The <code><dfn method for=PeriodicSyncManager title="getTags()">getTags()</dfn></code> method when invoked, MUST return [=a new promise=] |promise| and run the following steps [=in parallel=]:
```
Comment (rayankans) [inline spec/PeriodicBackgroundSync-index.bs]: This is racy, a parallel queue will solve this. I linked to it in another review

File: spec/PeriodicBackgroundSync-index.bs (hunk 2 above)
Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs]: done

File: spec/PeriodicBackgroundSync-index.bs (hunk 3 above)
Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs]: what does in do? I thought it's plain English.
Example text here (see number 7):
https://wicg.github.io/BackgroundSync/spec/#sync-manager-interface

Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs]: done

File: spec/PeriodicBackgroundSync-index.bs (hunk 4 above)
Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs]: In parallel, which I'm suggesting here, links to the same concept:
https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel

Comment (rayankans) [reply spec/PeriodicBackgroundSync-index.bs]: Yes, it proceeds to explain how to use parallel queues to get rid of race conditions (https://html.spec.whatwg.org/multipage/infrastructure.html#parallel-queue).

If you use the same parallel queue for all the algorithms you can also get rid of the re-registered while firing state.

File: spec/PeriodicBackgroundSync-index.bs (hunk 3 above)
Comment (rayankans) [reply spec/PeriodicBackgroundSync-index.bs]: Infra spec example uses `of`: https://infra.spec.whatwg.org/#list-iterate

It also explains how these concepts are closely related to the JavaScript language. In JS, `of` is used to iterate over lists, whereas `in` is used to iterate over an object's properties.

**Review by rayankans — APPROVED** [review]
Approving after offline discussion

File: spec/PeriodicBackgroundSync-index.bs (hunk 4 above)
Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs]: done.
Discussed offline why the reregisteredwhileFiring state is still necessary.

File: spec/PeriodicBackgroundSync-index.bs (hunk 3 above)
Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs]: done.

**Pull request merged**